app.py

```diff
@@ -0,0 +1,210 @@
+        help="Text to speak.",
+        value="The expressiveness of autoregressive transformers is literally nuts! I absolutely adore them.",
+    )
+
+    voices = os.listdir("tortoise/voices") + ["random"]
+    voices.remove("cond_latent_example")
+    voice = st.selectbox(
+        "Voice",
+        voices,
+        help="Selects the voice to use for generation. See options in voices/ directory (and add your own!) "
+        "Use the & character to join two voices together. Use a comma to perform inference on multiple voices.",
+        index=len(voices) - 1,
+    )
+    preset = st.selectbox(
+        "Preset",
+        (
+            "single_sample",
+            "ultra_fast",
+            "ultra_fast_old",
+            "fast",
+            "standard",
+            "high_quality",
+        ),
+        help="Which voice preset to use.",
+        index=1,
+    )
+    with st.expander("Advanced"):
+        col1, col2 = st.columns(2)
+        with col1:
+            """#### Model parameters"""
+            candidates = st.number_input(
+                "Candidates", help="How many output candidates to produce per-voice.", value=3
+            )
+            sampler = st.radio(
+                "Sampler",
+                SAMPLERS,
+                help="override the sampler used for diffusion (default depends on preset)",
+                index=1
+            )
+            steps = st.number_input(
+                "Steps",
+                help="Override the steps used for diffusion (default depends on preset)",
+                value=100,
+            )
+            seed = st.number_input(
+                "Seed", help="Random seed which can be used to reproduce results.", value=-1
+            )
+            if seed == -1:
+                seed = None
+            """#### Directories"""
+            output_path = st.text_input(
+                "Output Path", help="Where to store outputs.", value="results/"
+            )
+            model_dir = st.text_input(
+                "Model Directory",
+                help="Where to find pretrained model checkpoints. Tortoise automatically downloads these to .models, so this"
+                "should only be specified if you have custom checkpoints.",
+                value=MODELS_DIR,
+            )
+
+
+
+        with col2:
+            """#### Optimizations"""
+            high_vram = not st.checkbox(
+                "Low VRAM",
+                help="Re-enable default offloading behaviour of tortoise",
+                value=True,
+            )
+            half = st.checkbox(
+                "Half-Precision",
+                help="Enable autocast to half precision for autoregressive model",
+                value=False,
+            )
+            kv_cache = st.checkbox(
+                "Key-Value Cache",
+                help="Enable (partially broken) kv_cache usage, leading to drastic speedups but worse memory usage + results",
+                value=False,
+            )
```

Comment:
i don't think this should be disabled by default but i'll leave it there until someone else protests